### PR TITLE
pfunit: add v4.6.3, clean up recipe

### PR DESCRIPTION
**Pull request by alalazo**
Modifications:
- [x] Add v4.6.3
- [x] Deprecate v3.X
- [x] Removed `shared` variant (it seems only static builds are officially supported)
- [x] Fixed a few inconsistencies in argument names and values
- [x] Don't set unneeded environment variables

### CODE REVIEW

**Comment by alalazo** [comment]
@spackbot maintainers

**Comment by mathomp4** [comment]
@alalazo This is a bit of a change! I'll try and test it out soon

**Comment by alalazo** [comment]
Pipelines are failing on `develop` due to `pfunit@3.3` which is needed by `flecsi@1.4.2`, see:
- https://gitlab.spack.io/spack/spack/-/jobs/5540096

This is not appearing here because, since deprecating `pfunit@:3`  drives `clingo` to select a newer `flecsi@2.1.0` which doesn't depend on `pfunit`. Before `flecsi@1.4.2` was selected, since 1.4.2 is the preferred version.

@eugeneswalker @rspavel @ktsai7 @rbberger Is that a viable solution? Asking since `flecsi` seems quite a peculiar package, where the preferred version comes from a fork that is a couple of years behind the main repository.

**Comment by eugeneswalker** [comment]
> This is not appearing here because, since deprecating `pfunit@:3` drives `clingo` to select a newer `flecsi@2.1.0` which doesn't depend on `pfunit`. Before `flecsi@1.4.2` was selected, since 1.4.2 is the preferred version.
> 
> @eugeneswalker @rspavel @ktsai7 @rbberger Is that a viable solution? Asking since `flecsi` seems quite a peculiar package, where the preferred version comes from a fork that is a couple of years behind the main repository.

I don't know much about this package, other than we need to include some version of it in E4S. Whichever version that is or should be, I have to defer to those with more knowledge about how it is used. I'm fine with whatever works. But also, we don't want this to be holding up PRs, so if using a newer version makes things work here, I like that idea.

**Comment by haampie** [comment]
I'm gonna merge since it blocks other PRs, if someones doesn't agree feel free to follow up with a different solution.

**Review by haampie — APPROVED** [review]

**Comment by alalazo** [comment]
@mathomp4 Feel free to post here any follow up changes you'd need. I'll try to submit a PR for them, in case, asap.

**Pull request merged**
